src/Converter/LineItemsConverter.php

```diff
@@ -18,44 +18,119 @@
-                $variant !== null ? $variant->getCode() : null
+                $variant !== null ? $variant->getCode() : null,
+                $this->taxRateProvider->provide($orderItem)
             ));
         }
 
-        /** @var AdjustmentInterface $shippingAdjustment */
-        foreach ($shippingAdjustments as $shippingAdjustment) {
-            $lineItems->add(new LineItem(
-                $shippingAdjustment->getLabel(),
-                1,
-                $shippingAdjustment->getAmount(),
-                $shippingAdjustment->getAmount(),
-                0,
-                $shippingAdjustment->getAmount()
-            ));
+        return $lineItems;
+    }
+
+    private function getShipmentTaxAdjustment(ShipmentInterface $shipment): ?AdjustmentInterface
+    {
+        $taxAdjustments = $shipment->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT);
+        if ($taxAdjustments->isEmpty()) {
+            return null;
         }
 
-        return $lineItems;
+        if ($taxAdjustments->count() > 1) {
+            throw MoreThanOneTaxAdjustment::occur();
+        }
+
+        /** @var AdjustmentInterface $taxAdjustment */
+        $taxAdjustment = $taxAdjustments->first();
+
+        return $taxAdjustment;
+    }
+
+    private function getShipmentPromotionsAdjustmentAmount(ShipmentInterface $shipment): int
+    {
+        $promotionAdjustments = $shipment->getAdjustments(AdjustmentInterface::ORDER_SHIPPING_PROMOTION_ADJUSTMENT);
+
+        if ($promotionAdjustments->isEmpty()) {
+            return 0;
+        }
+
+        $promotionAdjustmentsAmount = 0;
+
+        foreach ($promotionAdjustments as $promotionAdjustment) {
+            $promotionAdjustmentsAmount = $promotionAdjustmentsAmount + $promotionAdjustment->getAmount();
+        }
+
+        return $promotionAdjustmentsAmount;
+    }
+
+    private function addLineItemForShipment(ArrayCollection $lineItems, OrderInterface $order): void
+    {
+        /** @var ShipmentInterface $shipment */
+        $shipment = null;
+
+        if (!$order->getShipments()->isEmpty()) {
+            $shipment = $order->getShipments()->first();
+        }
+
+        Assert::notNull($shipment);
+        Assert::isInstanceOf($shipment, AdjustableInterface::class);
+
+        $promotionAdjustmentsAmount = $this->getShipmentPromotionsAdjustmentAmount($shipment);
+
+        /** @var AdjustmentInterface $shippingAdjustment */
+        $shippingAdjustment = $shipment->getAdjustments(AdjustmentInterface::SHIPPING_ADJUSTMENT)->first();
+
+        $taxAdjustment = $this->getShipmentTaxAdjustment($shipment);
+        $taxAdjustmentAmount = $taxAdjustment !== null ? $taxAdjustment->getAmount() : 0;
+
+        /** @var string|null $label */
+        $label = $shippingAdjustment->getLabel();
+        Assert::notNull($label);
+
+        $lineItems->add(new LineItem(
+            $label,
+            1,
+            $promotionAdjustmentsAmount + $shippingAdjustment->getAmount(),
```

Comment:
`$promotionAdjustmentsAmount + $shippingAdjustment->getAmount()` can be extracted to variable `$netAmount`